flytekit/types/structured/structured_dataset.py

```diff
@@ -297,34 +320,33 @@ def to_literal(
 
         # Otherwise assume it's a dataframe instance. Wrap it with some defaults
         fmt = self.DEFAULT_FORMATS[python_type]
-        protocol = self.DEFAULT_PROTOCOLS[python_type]
+        protocol = protocol_prefix(ctx.file_access.get_random_remote_path())
```

Comment:
Remove - we shouldn't have to call file_access in this file.  If we do then I'm missing something and the interface is wrong.